src/middlewared/middlewared/plugins/disk.py

```diff
@@ -396,26 +392,11 @@ async def sed_unlock(self, disk_name, disk=None, _advconfig=None):
 
         # Try ATA Security if SED was not unlocked and its not locked by OPAL
         if not unlocked and not locked:
-            cp = await run('camcontrol', 'security', devname, check=False)
-            if cp.returncode == 0:
-                output = cp.stdout.decode()
-                if RE_CAMCONTROL_DRIVE_LOCKED.search(output):
-                    locked = True
-                    cp = await run(
-                        'camcontrol', 'security', devname,
-                        '-U', _advconfig['sed_user'],
-                        '-k', password,
-                        check=False,
-                    )
-                    if cp.returncode == 0:
-                        locked = False
-                        unlocked = True
-                else:
-                    locked = False
+            locked, unlocked = await self.middleware.call('disk.unlock_ata_security', devname, _advconfig, password)
 
         if unlocked:
             try:
-                # Disk needs to be retasted after unlock
+                # Disk needs to be retested after unlock
```

Comment:
It was not a typo, it's really `GEOM retaste`. Is it needed on Linux by the way?